app/src/lib/api.ts

```diff
@@ -292,10 +292,10 @@ export class API {
         private: private_,
       })
 
-      return parsedResponse<IAPIRepository>(response)
+      return await parsedResponse<IAPIRepository>(response)
     } catch (e) {
-      log.warn(`createRepository: failed with endpoint ${this.endpoint}`, e)
-      throw e
+      log.error(`createRepository: failed with endpoint ${this.endpoint}`, e)
+      throw new Error(`Unable to publish repository. Please check you have an internet connection and try again.`)
```

Comment:
We're not really handling any `4XX` errors explicitly (e.g. #1868 was a `422` but the PR that fixed it addressed the JSON parsing rather than the underlying logic).

I could go either way on whether we handle that here or do in a follow-up PR, but I do like that we're not rethrowing the original error here at least 🤷‍♂️ ...